oap-server/server-starter/src/main/resources/otel-oc-rules/k8s-instance.yaml

```diff
@@ -0,0 +1,23 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+filter: "{ tags -> tags.job_name in [ 'kubernetes-cadvisor', 'kube-state-metrics' ] }" # The OpenTelemetry job name
+expSuffix: |-
+  service(['cluster' , 'service'], '::', Layer.K8S_SERVICE)
+  .instance(['cluster', 'service'], '::', ['pod'], '', Layer.K8S_SERVICE, { tags -> ['pod': tags.pod, 'namespace': tags.namespace] })
+metricPrefix: k8s_service_instance
+metricsRules:
+  - name: pod_instance_status
+    exp: kube_pod_status_phase.retagByK8sMeta('service' , K8sRetagType.Pod2Service , 'pod' , 'namespace').tagNotEqual('service' , '').sum(['cluster', 'namespace', 'service' , 'pod'])
```

Comment:
This is a demo of how the extractor works (and will be used in on demand logs), my concern here is that we have to generate an unused instance metrics here, I'm trying to add a drop mechanism in MAL so that we can leverage the "side effect" but not store the metrics, would like to know what do you think @wu-sheng @wankai123

Reply:
A little curious, why don't we get the pod meta info from the K8s API server directly?

Reply:
@wankai123 What do you mean? This is for a streaming to add instance to represent a pod.

Reply:
Oh, I see. This is not for mesh, right?

Reply:
I think we don't expose enough API for this, no matter what we do to drop, it is a kind of hijacking.

I don't have much concern about this metric actually. This just adds an extra 1% metric :)

Reply:
> I think we don't expose enough API for this, no matter what we do to drop, it is a kind of hijacking.
> 
> I don't have much concern about this metric actually. This just adds an extra 1% metric :)

Hi, I didn't mean to drop the metrics in the streaming process, I meant to drop the metrics just before MAL sends those metrics, see my changes in https://github.com/apache/skywalking/pull/9151/commits/1db9e71c5401a9a4b91536e7b3b397cd7a6769c8

Reply:
see https://github.com/apache/skywalking/pull/9151#discussion_r886266807 

We could create an instance(pod-oriented) dashboard to make this clear. But for now, even without them, I feel it is fine.